### ISSUE TITLE
Rule proposal: no-event-handler-return-values

### ISSUE THREAD

**Issue by ljharb** [yannickcr/eslint-plugin-react#692]
When an `onFoo` handler is created inline, a la an arrow function, this rule should require that the function have no return value.

```js
// bad
<Foo onChange={x => y} />
<Foo onChange={x => { return y; }} />
<Foo onChange={function (x) { return y; }} />
const foo = x => y;
<Foo onChange={foo} />

// good
<Foo onChange={x => { y; }} />
<Foo onChange={function (x) { y; }} />
```

The goal is to prevent event handlers from returning `false`, so that they instead must explicitly call `preventDefault` or `stopPropagation` as desired.

This would ignore anything starting with "on" that wasn't a React-supported event handler, and would ignore a value that wasn't a function created in the same file.
